multiplexer/abci/multiplexer.go

```diff
@@ -52,6 +53,9 @@ const (
 type Multiplexer struct {
 	logger log.Logger
 	mu     sync.Mutex
+	// done is a hack to prevent the multiplexer from catching the above mutex twice and avoiding mu.TryLock
+	// as that could result in an accidental crash.
+	done atomic.Bool
```

Comment:
per comments, this is a hack and a longer term solution likely involves a refactor. I could have used TryLock on the mutex, but I was worried that that was not explicit enough and maybe could be hit if we accidentally call the mutex twice